src/renderer/src/components/LeaveProjectModal/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { Dialog } from '@mui/material'
+import { useState } from 'react'
+import { LeaveProjectConfirmation } from './LeaveProjectConfirmation'
+import { DeleteData } from './DeleteData'
+import { SuccessfulLeave } from './SuccessfulLeave'
+
+type LeaveProjectModalProps = {
+  isOpen: boolean
+  closeModal: () => void
+}
+
+export const LeaveProjectModal = ({ isOpen, closeModal }: LeaveProjectModalProps) => {
+  function handleCloseDialog(event?: Record<string, never>, reason?: 'escapeKeyDown' | 'backdropClick') {
+    if (reason === 'backdropClick' || reason === 'escapeKeyDown') return
+    closeModal()
+  }
+
+  const [modalContent, setModalContent] = useState<'confirmation' | 'deleteData' | 'successfulLeave'>(
+    'confirmation',
+  )
+
+  return (
+    <Dialog open={isOpen} onClose={handleCloseDialog} maxWidth="xl" fullWidth>
+      {modalContent === 'confirmation' && (
+        <LeaveProjectConfirmation
+          moveToDeleteDataContent={() => {
+            setModalContent('deleteData')
+          }}
+        />
```

Comment:
this is missing a `close` prop. pressing the `Cancel` button in this state causes the app to crash

Reply:
actually i think what's happening is you're calling [`window.close`](https://developer.mozilla.org/en-US/docs/Web/API/Window/close), which causes the app window to close without having a way to restore it 😅 . I generally avoid calling props `close` because there's an implicit `close` global, so TS won't always help you in the way you expect.

Reply:
no, that is suppose to be from props, not `window.close`

Reply:
right, but you're not providing a `close` prop here so `LeaveProjectConfirmation` is calling `close` [here](https://github.com/digidem/mapeo-desktop-shell/blob/696d980fe083928f4de575f3e81851acd93e6275/src/renderer/src/components/LeaveProjectModal/LeaveProjectConfirmation.tsx#L124), which is the same as `window.close` because of how the browser works

Reply:
https://github.com/digidem/mapeo-desktop-shell/pull/56/commits/f5d14d3ac957f1e7c82fa841e3608d8a288a3405